azurerm/resource_arm_servicebus_namespace_authorization_rule.go

```diff
@@ -0,0 +1,217 @@
+
+			if rights.Contains(string(servicebus.Manage)) {
+				if !rights.Contains(string(servicebus.Listen)) || !rights.Contains(string(servicebus.Manage)) {
+					return fmt.Errorf("In order to allow the 'Manage' right - both the 'Listen' and 'Send' rights must be present")
+				}
+			}
+
+			return nil
+		},
+	}
+}
+
+func resourceArmServiceBusNamespaceAuthorizationRuleCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).serviceBusNamespacesClient
+	ctx := meta.(*ArmClient).StopContext
+
+	log.Printf("[INFO] preparing arguments for AzureRM ServiceBus Namespace Authorization Rule creation.")
+
+	name := d.Get("name").(string)
+	resGroup := d.Get("resource_group_name").(string)
+	namespaceName := d.Get("namespace_name").(string)
+
+	parameters := servicebus.SBAuthorizationRule{
+		Name: &name,
+		SBAuthorizationRuleProperties: &servicebus.SBAuthorizationRuleProperties{
+			Rights: expandServiceBusAuthorizationRuleRights(d),
+		},
+	}
+
+	_, err := client.CreateOrUpdateAuthorizationRule(ctx, resGroup, namespaceName, name, parameters)
+	if err != nil {
+		return err
+	}
+
+	read, err := client.GetAuthorizationRule(ctx, resGroup, namespaceName, name)
+	if err != nil {
+		return err
+	}
+
+	if read.ID == nil {
+		return fmt.Errorf("Cannot read ServiceBus Namespace Authorization Rule %s (resource group %s) ID", name, resGroup)
+	}
+
+	d.SetId(*read.ID)
+
+	return resourceArmServiceBusNamespaceAuthorizationRuleRead(d, meta)
+}
+
+func resourceArmServiceBusNamespaceAuthorizationRuleRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).serviceBusNamespacesClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resGroup := id.ResourceGroup
+	namespaceName := id.Path["namespaces"]
+	name := id.Path["AuthorizationRules"] //this is slightly different this topic, authorization is capitalized
+
+	resp, err := client.GetAuthorizationRule(ctx, resGroup, namespaceName, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error making Read request on Azure ServiceBus Namespace Authorization Rule %s: %+v", name, err)
+	}
+
+	keysResp, err := client.ListKeys(ctx, resGroup, namespaceName, name)
+	if err != nil {
+		return fmt.Errorf("Error making Read request on Azure ServiceBus Namespace Authorization Rule List Keys %s: %+v", name, err)
+	}
+
+	d.Set("name", name)
+	d.Set("namespace_name", namespaceName)
+	d.Set("resource_group_name", resGroup)
+
+	if err := d.Set("rights", flattenServiceBusAuthorizationRuleRights(resp.Rights)); err != nil {
+		return fmt.Errorf("Error flattening rights: %+v", err)
+	}
+
+	d.Set("primary_key", keysResp.PrimaryKey)
+	d.Set("primary_connection_string", keysResp.PrimaryConnectionString)
+	d.Set("secondary_key", keysResp.SecondaryKey)
+	d.Set("secondary_connection_string", keysResp.SecondaryConnectionString)
```

Comment:
can we access rights -> secondary_connection_string via the `[XX]Properties` object & nil-check them?

Reply:
There are no properties object returned by list keys.